basic_pipeline_extension/01.0_Bin.md

```diff
@@ -0,0 +1,106 @@
+# Bin
+
+A Membrane's bin is a container for elements, which allows for creating reusable groups of elements.
+Bin is similar to a pipeline in that it consists of linked elements. Such bin can then be placed inside a pipeline and linked with other elements. Bins can also be nested within one another.
```

Comment:
Hmm, in fact `Bin` does not `use Membrane.Element.Base`, therfore I wouldn't refer to it as an element. To be precise, a bin is a container for element and other bins. I know you have mentioned that bins can be nested withing one another, but I would stick to a naming convention that elements are: filters/sources/sinks/endpoints

Reply:
The fact, that in some cases bin behaves like an element (it can be put in the pipeline, it has pads) and in other cases it behaves like a pipeline (as you have described) might be worth mentioning

Reply:
What word would you use, that describes both bins and elements? component, entity, item?

Reply:
Well, in Pipeline's API it is refered as "children" (meaning - "Pipeline's children"), but I am not sure if it fits well here, WDYT?

Reply:
I'll go with `entity`